### PR TITLE
Add source code overview doc

### PR DESCRIPTION
**Pull request by caipengbo**
I added the Chinese version of the source code overview to help developers understand the code logic of KVRocks more quickly.

If there is no problem with the CR, I will translate it into English so that it can have a wider impact.

### CODE REVIEW

File: docs/source-code-overview.CN.md [hunk 1]
```diff
@@ -0,0 +1,209 @@
+# KVRocks源码概览
+# 整体流程
+
+1. 加载 KVRocks 配置(解析配置文件，构建 `Config` )
+2. 初始化存储引擎 `Engine::Storage`， 并打开
```
Comment (git-hulk) [inline docs/source-code-overview.CN.md]:
```suggestion
2. 初始化并打开存储引擎 `Engine::Storage` 
```

File: docs/source-code-overview.CN.md [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+
+## 复制线程
+
+执行主从复制相关的逻辑，见：[metadata-design](./metadata-design.md)
+
+# 命令执行
+
+## 编码
+
+见：[replication-design](./replication-design.md)
```
Comment (git-hulk) [inline docs/source-code-overview.CN.md]: Refer the wrong link

**Review by git-hulk — COMMENTED** [review]
Huge thanks @caipengbo for writing this.

BTW, we can have a whitespace between Chinese and English words (or numbers) to make the doc more readable, also not mix English and Chinese comma.

**Comment by caipengbo** [comment]
Thank you @git-hulk , I will modify those problems and update the contents based on latest source code.

File: docs/source-code-overview.CN.md [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+- redis_request.h, redis_request.cc
+
+主要用来解析 eventbuffer 中的数据, 解析成 Redis 命令，并执行
+
+`Request::Tokenize()`：
+
+将客户端传来的数据（从 `eventbuffer` 中）读出，分隔成 Token
+
+`Request::ExecuteCommands()`：
+- 调用 `LookupCommand()` 查找命令表
```
Comment (ShooterIT) [inline docs/source-code-overview.CN.md]: we have updated this implementation, moved it to `connection`.
`LookupCommand` also is updated to another one.

File: docs/source-code-overview.CN.md [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+
+## TaskRunner 线程池
+
+涉及到的文件：
+
+- task_runner.h, task_runner.cc
+
+是一个线程池，有任务队列，用来存储异步的任务(Task)，当前异步的任务有：
+
+- `Server::AsyncCompactDB()`: 被 `compact` 命令、`Server::cron` 创建任务
```
Comment (ShooterIT) [inline docs/source-code-overview.CN.md]: 去掉`被`吧，how about改为xxx调用

File: docs/source-code-overview.CN.md [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+- `Server::AsyncBgsaveDB()`: 被 `bgsave` 命令、`Server::cron` 调用
+- `Server::AsyncPurgeOldBackups()`: 被 `flushbackup` 命令、`Server::cron` 调用
+- `Server::AsyncScanDBSize()`: 被 `dbsize` 命令
```
Comment (ShooterIT) [inline docs/source-code-overview.CN.md]: `dbsize` 命令调用

File: docs/source-code-overview.CN.md [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+4. 终止key
+
+对满足以下条件的SST文件进行手动Compaction: 
+
+1. 创建超过两天的SST文件  
+2. 删除key占比多的SST文件
+
+## 复制线程
+
+执行主从复制相关的逻辑，见：[metadata-design](./metadata-design.md)
```
Comment (ShooterIT) [inline docs/source-code-overview.CN.md]: metadata-design？

File: docs/source-code-overview.CN.md (hunk 3 above)
Comment (caipengbo) [reply docs/source-code-overview.CN.md]: Ok, I will update it right now.

File: docs/source-code-overview.CN.md [hunk 7]
```diff
@@ -0,0 +1,221 @@
+# KVRocks源码概览
```
Comment (ShooterIT) [inline docs/source-code-overview.CN.md]:
```suggestion
# Kvrocks源码概览
```
It really is a big trouble for us how to spell kvrocks, but we currently, we still prefer to use **Kvrocks** when we need upper case.
Please also fix the following  `KVRocks`.

**Review by karelrooted — DISMISSED** [review]

File: docs/source-code-overview.CN.md (hunk 7 above)
Comment (caipengbo) [reply docs/source-code-overview.CN.md]: Done

**Review by git-hulk — DISMISSED** [review]

File: docs/source-code-overview.CN.md [hunk 8]
```diff
@@ -0,0 +1,221 @@
+# Kvrocks 源码概览
```
Comment (ShooterIT) [inline docs/source-code-overview.CN.md]:
```suggestion
```
I prefer to remove that, because that is title.

File: docs/source-code-overview.CN.md [hunk 9]
```diff
@@ -0,0 +1,221 @@
...
+
+获取 SST 属性
+
+`CompactOnExpiredCollector` 通过继承 `rocksdb::TablePropertiesCollector` 实现自定义 SST 属性，然后实现相应的工厂类 `CompactOnExpiredTableCollectorFactory`，将工厂类通过 `rocksdb::ColumnFamilyOptions` 传递给存储引擎。
+
+## 主从复制线程
+
+执行主从复制相关的逻辑，具体见: [replication-design](./replication-design.md)
+
+# 命令执行
```
Comment (ShooterIT) [inline docs/source-code-overview.CN.md]:
```suggestion
# 命令实现
```

**Review by ShooterIT — APPROVED** [review]

**Review by git-hulk — APPROVED** [review]

**Pull request merged**